### DISCUSSION TITLE
Brightness always on 100% after restart

### DISCUSSION THREAD

**Discussion by abenezerw** [prasanthrangan/hyprdots#724] [Q&A] [answered]
Apologies, you’re right about cluttering previous thread. 
the brightness -s didn’t work and set to 100. 
but the second set of codes work fine and do bring back the initial level. 
![IMG_1014](https://github.com/prasanthrangan/hyprdots/assets/46224293/01b83acd-d8ce-4917-b221-3ffb2fcd1f76)
![IMG_1015](https://github.com/prasanthrangan/hyprdots/assets/46224293/a0f46bbc-503a-4e00-8851-2c35aa8af3df)
Btw there’s also another issue I don’t know if you can see but there is a thin line under the waybar which always shows a wallpaper that’s different from the above. I dont know if it’s scaling issues or the next wallpaper. Here’s another image for reference. 
![IMG_1016](https://github.com/prasanthrangan/hyprdots/assets/46224293/26f6211e-bed0-459f-ad22-c3fb995e6ef3)
@kRHYME7

**Comment by kRHYME7** [comment, marked as answer]
For wallpaper issues 
Put these lines in the hypland.conf

```sh
misc {
    disable_hyprland_logo = true
    disable_splash_rendering = true
}
```


I guess this is due to some updates.

**Comment by kRHYME7** [comment]
> but the second set of codes work fine and do bring back the initial level.

About the brightness


add this line:  ``` brightnessctl g > ~/.cache/hyprdots/brightness #? Saves Brightness to cache ```
to at the end of ``` ~/.config/hypr/scripts/brightnesscontrol.sh   ````


Then add this line : ``` exec-once = brightnessctl s $(cat ~/.cache/hyprdots/brightness)   # Restore Brightness ```
in your ``` ~/.config/hypr/userprefs.conf ```

PS. This is only a workaround try to read on the archwiki how to handle this brightness

**Reply by abenezerw** [reply]
yeah, this should sort it out. but i think i saw in the archwiki about how this could be a result of not having proprietary NVIDIA. some other distros also have people with similar issues.

**Comment by abenezerw** [comment]
Maybe this would help with something. @kRHYME7 
I’m back. Haha 2 full days of troubleshooting. So 
On issue #710 
Was mentioned how the culprit to low res vscode and Spotify could be nouveau 
And wanted to game so I managed to install Nvidia-dkms and delete nouveau 
Which solved most my problems, primarily the brightness always at 100% on restart. 
but now I tried to do the multi-gpu thing to always use amd for Hyprland first but in the waybar, I still see the primary gpu listed as Nvidia instead of amd despite adding that env= card1 line. Here is reference with more info. Last line in env section. 
![IMG_1059](https://github.com/prasanthrangan/hyprdots/assets/46224293/d5800b30-2616-4b43-a0fd-e7be34ce59b0)

**Comment by kRHYME7** [comment]
> I still see the primary gpu listed as Nvidia instead of amd despite adding that env= card1 line. Here is reference with more info. Last line in env section.


Yow, that GPU script is not related to any environments on Hyprland (It is an independent script).   If you have multi-GPU, to toggle this we should do it manually. 
On the PR #643  I added the last line of  ``` ~/.config/waybar/modules/gpuinfo.jsonc ```
this
```  "on-click": "~/.config/hypr/scripts/gpuinfo.sh --toggle", ```

so it would look like like this now:
```
"custom/gpuinfo": {
    "exec": " ~/.config/hypr/scripts/gpuinfo.sh",
    "return-type": "json",
    "format": " {}",
    "interval": 5, // once every 5 seconds
    "tooltip": true,
    "max-length": 1000,
    "on-click": "~/.config/hypr/scripts/gpuinfo.sh --toggle",
},
```

Be sure to restart waybar or toggle the change waybar layout to refresh


to test it out you can run toggle
``` ~/.config/hypr/scripts/gpuinfo.sh --toggle ```

more info at this #643   Please read the PR.

**Comment by abenezerw** [comment]
Yup. That worked with the terminal though. ~/.config/hypr/scripts/gpuinfo.sh --toggle When I ran that. 
actually clicking the button doesn’t switch between though.

**Reply by kRHYME7** [reply]
did you added this ? 
    "on-click": "~/.config/hypr/scripts/gpuinfo.sh --toggle",
Will take  effect every 5 seconds.

**Reply by abenezerw** [reply]
![image](https://github.com/prasanthrangan/hyprdots/assets/46224293/3d8623ad-513c-40de-b034-ec8d5edf9026)
Yup. But I have to manually do that from the command line to toggle between them. Maybe my version of gpu info.sh?

**Reply by abenezerw** [reply]
Just doing a click makes it go away. Even if I waited for 5 secs.

**Reply by kRHYME7** [reply]
what is the output off

``` cat /tmp/hyprdots-gpuinfo-query* ```

**Reply by abenezerw** [reply]
![image](https://github.com/prasanthrangan/hyprdots/assets/46224293/0be20c3c-0b37-4bc6-8951-c19f296e344a)

**Reply by kRHYME7** [reply]
nice it is working as intended  ~how about  cat /.config/waybar/modules/gpuinfo.jsonc~

~Also you can try to update that GPU script incase. I pushed a recent one~

Strange Try the Latest commit . 


# And Also do not forget to restart waybar

**Reply by abenezerw** [reply]
i manually tried downloading this one from this link 
https://github.com/prasanthrangan/hyprdots/pull/643/files#diff-208d186a74bb2c3911ce07b6d91cf3c8af7348856609f5353544e88ab02c007e
but and forgot how to kill waybar. but i restarted the system and the gpu temp icon isnt there entirely. so i reverted to the original jsonc file.
wait maybe i should do the chmod thing you said last time?

**Reply by kRHYME7** [reply]
In whatever case use this for the module
```
"custom/gpuinfo": {
    "exec": " ~/.config/hypr/scripts/gpuinfo.sh",
    "return-type": "json",
    "format": " {}",
    "interval": 5, // once every 5 seconds
    "tooltip": true,
    "max-length": 1000,
    "on-click": "~/.config/hypr/scripts/gpuinfo.sh --toggle",
},

```

then 
~run : ``` killall waybar || nohup waybar ```~ 
### This wont work Just toggle waybar by changing layout

ad: 
run ``` Hyprdots show binds ```
then search the toggle waybar I guess

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/da79e7e7-52e4-4eda-9bd5-66acbc87c788)

**Comment by abenezerw** [comment]
So I did all of that and when I pressed the button to switch waybar theme it disappeared entirely, and when I try to run waybar from command line it shows an error in a line with a missing , or } but I checked that file and it’s similar to what you typed. 
![image](https://github.com/prasanthrangan/hyprdots/assets/46224293/7b2f34af-1fdb-4d4d-b6d2-97029d6a88dd)

**Reply by kRHYME7** [reply]
try to change waybar layout in my case here is the keybind 

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/397b32bd-d7b8-4660-83c9-6fd7ad10c945)

**Reply by abenezerw** [reply]
That worked. But now the some waybars don’t show up until I keep cycling. I’m assuming they’re the ones with the gpu info in it.

**Reply by kRHYME7** [reply]
>Assuming the ./gpuinfo.sh is working 
try to use the default config from the original Repository then see if it works then 
add this line 
    "on-click": "~/.config/hypr/scripts/gpuinfo.sh --toggle"

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/6796cd6c-be14-4c65-b055-8ecbc1d79e75)


### My Bad Add " , " a comma like this  
``` 
    "max-length": 1000, 
 ```

**Reply by abenezerw** [reply]
Works! When I reverted to original file from repo and added the onclick thing too. 
a little buggy because it fades away after initial click instead of just staying there and switching to next gpu info but if you just swipe the mouse away and hover over it again. You can see the info for the next gpu.

**Reply by abenezerw** [reply]
Thank you

**Reply by kRHYME7** [reply]
I guess that is how waybar tooltip works.

**Reply by kRHYME7** [reply]
@abenezerw 

>but now I tried to do the multi-gpu thing to always use amd for Hyprland first but in the waybar, I still see the primary gpu listed as Nvidia instead of amd despite adding that env= card1 line. Here is reference with more info. Last line in env section.


Hello! I realized we can add a feature, if a user declared the gpu in the env on hyprland like you did, the script will assume that GPU to be the primary GPU. Can I have favor, to test this? It is now on the PR also a review will be great!  Thanks

another favor what is the output for 

```  lspci | grep -E "VGA|3D" ```   I do not have AMD so I cannot test for AMD users, this would be very helpful.

**Reply by abenezerw** [reply]
There was oversight on my part about this. Because when I checked nvtop it wasn’t actually using Nvidia, it was just the script that shows Nvidia on 0% utilization, but the fix you told me to toggle amd, I saw that it’s actually using amd for Hyprland. I guess this is the problem you’re trying to fix. Here’s the pic you asked for. 
![image](https://github.com/prasanthrangan/hyprdots/assets/46224293/9decd3dc-f0ea-4ab8-91e5-19cfda39782a)
What do you mean review? Great job btw. Definitely loving my pc now.

**Reply by kRHYME7** [reply]
@abenezerw  I made some oversight here I meant output for ``` lspci -nn | grep -E "VGA|3D" ``` I forgot the -nn part.  Thank you for baring.

**Reply by abenezerw** [reply]
Same thing. 
![image](https://github.com/prasanthrangan/hyprdots/assets/46224293/337fe627-f6ac-49db-b461-9be4db896a50)
Oh wait it has the little code things in brackets after controller.
